plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsConnectorTableHandle.java

```diff
@@ -16,27 +16,55 @@
 import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.NotFoundException;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TableNotFoundException;
 
 @JsonTypeInfo(
         use = JsonTypeInfo.Id.NAME,
         property = "@type")
 @JsonSubTypes({
-        @JsonSubTypes.Type(value = SheetsNamedTableHandle.class, name = "named")
+        @JsonSubTypes.Type(value = SheetsNamedTableHandle.class, name = "named"),
+        @JsonSubTypes.Type(value = SheetsSheetTableHandle.class, name = "sheet")
 })
 public sealed interface SheetsConnectorTableHandle
         extends ConnectorTableHandle
-        permits SheetsNamedTableHandle
+        permits SheetsNamedTableHandle, SheetsSheetTableHandle
 {
     default boolean isNamedTable()
     {
         return this instanceof SheetsNamedTableHandle;
     }
 
+    default boolean isSheetTable()
```

Comment:
I don't think that the methods `isSheetTable()` etc. add to readability. I would just use explicit class checks and casts instead. I'm not sure we even need the enclosing interface.

Reply:
When [pattern matching ](https://openjdk.org/jeps/427)comes out of preview, the sealed interface permits exhaustive pattern matching on the implementing interfaces. 
 
I think the helper methods adds readability over lots of instanceof and casts everywhere, but readability is more subjective.

Can revert if you feel strongly about it.